ext/rugged/rugged_config.c

```diff
@@ -143,6 +143,53 @@ static VALUE rb_git_config_store(VALUE self, VALUE rb_key, VALUE rb_val)
 	return Qnil;
 }
 
+/*
+ *  call-seq:
+ *   cfg.add(key, value)
+ *
+ *  Store the given +value+ in the Config file, under the section
+ *  and name specified by +key+, without replacing any existing
+ *  entries. Value can be any of the following Ruby types:
+ *  +String+, +true+, +false+ and +Fixnum+.
+ *
+ *  The config file will be automatically stored to disk.
+ *
+ *    cfg.add('custom.multi.value', 'fix')
+ *    cfg.add('custom.multi.value', true)
+ *    cfg.add('custom.multi.value', 90)
+ */
+static VALUE rb_git_config_add(VALUE self, VALUE rb_key, VALUE rb_val)
+{
+	git_config *config;
+	const char *key;
+	int error;
+
+	Data_Get_Struct(self, git_config, config);
+	Check_Type(rb_key, T_STRING);
+
+	key = StringValueCStr(rb_key);
+
+	/*
+	 * "$^" is an unmatchable regexp: it will not match anything at all, so
+	 * all values will be considered new and will not replace any present value.
+	 */
+	if (RB_TYPE_P(rb_val, T_STRING)) {
+		error = git_config_set_multivar(config, key, "$^", StringValueCStr(rb_val));
+	} else if (RB_TYPE_P(rb_val, T_TRUE) || RB_TYPE_P(rb_val, T_FALSE)) {
+		error = git_config_set_multivar(config, key, "$^", (rb_val == Qtrue) ? "true" : "false");
+	} else if (FIXNUM_P(rb_val)) {
+		char str_value[32];
```

Comment:
I took this snippet from the [libgit2 source](https://github.com/libgit2/libgit2/blob/63307cbad7408b38693e73b74560a063da21f530/src/config.c#L627).  I'm not entirely sure this is the best way to do this conversion, but I figured I'd at least mimic the behavior.  There are some performance enhancements for some windows systems for `snprintf` that were not ported over (because I have no idea how that works).